references/classification/train_quantization.py

```diff
@@ -46,13 +64,20 @@ def main(args):
 
     print("Creating model", args.model)
     # when training quantized models, we always start from a pre-trained fp32 reference model
-    model = torchvision.models.quantization.__dict__[args.model](weights=args.weights, quantize=args.test_only)
+    if use_fx_graph_mode_quantization:
+        model = torchvision.models.__dict__[args.model](weights=args.weights)
```

Comment:
I suspect this change doesn't work for all cases. What if you want to perform inference (with `--test-only`) on a model trained with QAT? The way we do it here, we will end up evaluating the unquantized model.

Reply:
I think you're right. What we want here is to have something like the following:
```
model = torchvision.models.resnet50(ResNet50_QuantizedWeights.IMAGENET1K_FBGEMM_V1)
```
However, the way the unquantized models are set up today there is no way to pass in `ResNet50_QuantizedWeights` to `torchvision.models.resnet50`. I think one way forward is to make `ResNet50_QuantizedWeights` a subclass of `ResNet50_Weights`. I will look into this further.

Reply:
This is why I'm trying to figure out what are the steps for loading pre-trained quantized weights to a model. Can this be done directly to the original model without any conversions? Or some FX convert call is needed prior loading?

Reply:
I was unable to make `ResNet50_QuantizedWeights` inherit from `ResNet50_Weights` in the end because they are both enums. Instead I made FX mode go through the quantized model builders (`torchvision.models.quantization.resnet`) which are more extensible. Let's move this discussion to the main thread, where I added a comment to clarify this new approach.

Reply:
Quick answer to your question here is, yes, in `test_only` mode, we need to call `convert` before loading the weights, since they are quantized weights. This is true for both eager and FX. However, PTQ and QAT don't have this constraint.